### ISSUE TITLE
Deactivate method isn't called when user is moving between two child routers.

### ISSUE THREAD

**Issue by milunka** [aurelia/router#132]
I think that deactivate method isn't called on page for this situation:
- application has 3 routers defined as one master router with two child routers
- user moves from page in one child router to different page in different child router and deactivate method on page where user was at the begining isn't called.

Example of how child routers are configured:

**app.js**:

```
config.map([
  { route: '',  moduleId: './welcome',      nav: true, title:'Welcome' },
    { route: 'child1', moduleId: './child1', nav: true, title: 'Child 1' },
    { route: 'child2', moduleId: './child2', nav: true, title: 'Child 2' },
  ]);
```

**child1.js**

```
config.map([
  { route: '', moduleId: './page1', nav: true, title:'Page 1' },
  { route: 'page1a', moduleId: './page1a', nav: true, title:'Page 1a' },
]);
```

**child2.js**

```
config.map([
  { route: '', moduleId: './page2', nav: true, title:'Page 2' },
]);
```

When user is on _Page 1a_ and moves to _Page 2_ this is list of called events:
- child1.js - deactivate
- child2.js - activate
- page2.js - activate
- child1.js - detached
- page1a.js - detached
- child2.js - attached
- page2.js - attached

I think that as first method must be called deactivate method on _Page 1a_ and then all methods which are called now.

Demo of this problem: https://github.com/milunka/redirect-activate-problem

**Comment by brantwedel** [comment]
I am having this same problem in a slightly different scenario:

When a parent route changes with a replace strategy:
```js
determineActivationStrategy() { return activationStrategy.replace; }
```

deactivate does not get called on child VM's, but since `canDeactivate` *is* called in this scenario, I am using this workaround:

```
class VM {
    deactivate() {
        // deactivation logic, ensure this code does not have side effects if called twice
    }
    canDeactivate() {
        return this.deactivate() || true;
    }
}
```

**Comment by derekpitt** [comment]
Is this still an issue? I am seeing the same thing here too..

**Comment by heruan** [comment]
I confirm this, `deactivate` isn't called between VMs on the same navigation hierarchy.

**Comment by tomtomau** [comment]
Also echoing the same issue as @heruan

**Comment by heruan** [comment]
Will this possibly be addressed in the next release? Since now I was using @brantwedel trick, but I'm facing a scenario where I really need `deactivate` to be called *only if* `canDeactivate` returns true!

**Comment by EisenbergEffect** [comment]
@bryanrsmith Can you look at this with top priority?

**Comment by heruan** [comment]
Any news on this?

**Comment by EisenbergEffect** [comment]
No news. As always, if an issue is important to you, we welcome all PR. That's the best way to get a particular issue prioritized.

**Comment by MT-PL-SWF-Dev** [comment]
I think that this issue is similar to https://github.com/aurelia/router/issues/411
In that issue I suggested simple solution, still I'm not sure if this is  valid way

**Comment by qraynaud** [comment]
Was this issue also fixed by #411 ?
